### DISCUSSION TITLE
having the error shown in the minibuffer instead of function signatures

### DISCUSSION THREAD

**Discussion by CDitzel** [joaotavora/eglot#1104] [General]
`
 (with-eval-after-load 'eglot
   (add-to-list 'eglot-server-programs
                '((c-mode c++-mode cuda-mode)
                 . ("clangd-15"
                    "-j=8"
                    "--log=error"
                    "--malloc-trim"
                    "--background-index"
                    "--clang-tidy"
                    "--cross-file-rename"
                    "--completion-style=detailed"
                    "--pch-storage=memory"
                    "--header-insertion=never"
                    "--header-insertion-decorators=0"))))

			  (setq eldoc-echo-area-use-multiline-p 1)

`

I have this config which causes the minibuffer to occupy 1 line at all times only. However, by default, the function signature is shown there when point is over the function. I would like there to be shown the clangd errors, which right now I can only discover by hovering with the mouse over it. How can I change that?

**Comment by joaotavora** [comment]
Carsten Ditzel ***@***.***> writes:

> `
> (with-eval-after-load 'eglot
> (add-to-list 'eglot-server-programs
> '((c-mode c++-mode cuda-mode)
> . ("clangd-15"
> "-j=8"
> "--log=error"
> "--malloc-trim"
> "--background-index"
> "--clang-tidy"
> "--cross-file-rename"
> "--completion-style=detailed"
> "--pch-storage=memory"
> "--header-insertion=never"
> "--header-insertion-decorators=0"))))
>
> 		  (setq eldoc-echo-area-use-multiline-p 1)
>
> `
>
> I have this config which causes the minibuffer to occupy 1 line at all times only. However, by default, the function
> signature is shown there when point is over the function. I would like there to be shown the clangd errors, which right
> now I can only discover by hovering with the mouse over it. How can I change that?

You should configure eldoc-documentation-functions in
eglot-managed-mode-hook, adding in flymake-eldoc-function as one of its
elements.  I'm looking into exactly your problem and trying to change
the default value, but it's not so simple because reasons.
